bands/models.py

```diff
@@ -18,118 +18,168 @@ class Event:
 class Band(MagModel):
     group_id = Column(UUID, ForeignKey('group.id'))
     event_id = Column(UUID, ForeignKey('event.id'), nullable=True)
+    info = relationship('BandInfo', backref=backref('band', load_on_pending=True), uselist=False)
+    bio = relationship('BandBio', backref=backref('band', load_on_pending=True), uselist=False)
+    taxes = relationship('BandTaxes', backref=backref('band', load_on_pending=True), uselist=False)
+    stage_plot = relationship('BandStagePlot', backref=backref('band', load_on_pending=True), uselist=False)
+    panel = relationship('BandPanel', backref=backref('band', load_on_pending=True), uselist=False)
+    merch = relationship('BandMerch', backref=backref('band', load_on_pending=True), uselist=False)
+    charity = relationship('BandCharity', backref=backref('band', load_on_pending=True), uselist=False)
 
     payment = Column(Integer, default=0, admin_only=True)
     vehicles = Column(Integer, default=1, admin_only=True)
     estimated_loadin_minutes = Column(Integer, default=60, admin_only=True)
     estimated_performance_minutes = Column(Integer, default=60, admin_only=True)
 
+    @property
+    def all_badges_claimed(self):
+        return not any(a.is_unassigned for a in self.group.attendees)
+
+    @property
+    def estimated_performer_count(self):
+        return len([a for a in self.group.attendees if a.badge_type == c.GUEST_BADGE]) or 0
+
+    @property
+    def performance_minutes(self):
+        return self.event.minutes if self.event_id else self.estimated_performance_minutes
+
+    @property
+    def email(self):
+        return self.group.email
+
+    def completed(self, model):
```

Comment:
This is the new function that lets you check if steps are completed without printing out a stacktrace if the steps don't exist.

I still think it's more logical to put each "completed" step within the class it checks, and templates can still directly use that property, but this is ALSO useful for expansions in case someone forgets to add that property - it checks the ID instead. I tested this and made sure that it won't override a 'completed' property that exists.